datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -596,7 +596,7 @@ struct BufferedBatch {
     pub size_estimation: usize,
     /// The indices of buffered batch that failed the join filter.
     /// When dequeuing the buffered batch, we need to produce null joined rows for these indices.
-    pub join_filter_failed_idxs: HashSet<u64>,
+    pub join_filter_failed_map: HashMap<u64, bool>,
```

Comment:
Please comment what is key and value in the map

Reply:
Added comment.